src/windows/loadsave.c

```diff
@@ -154,9 +154,15 @@ rct_window *window_loadsave_open(int type, char *defaultName)
 	int includeNewItem;
 	rct_window* w;
 	_type = type;
-	_defaultName[0] = 0;
+	_defaultName[0] = '\0';
 
 	if (!str_is_null_or_empty(defaultName)) {
+		// From the scenario editor defaultName will be just the name, but when
+		// saving saved games it will contain the extension.
+		char *extension = (char*)path_get_extension(defaultName);
+		if (_stricmp(extension, ".sv6") == 0)
```

Comment:
For parks with a dot in the name it would remove the part behind it, because the extension is part of defaultName.